src/Modal/Modal.js

```diff
@@ -0,0 +1,81 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+
+import ModalHeader from "./ModalHeader";
+import Portal from "./Portal";
+
+class Modal extends Component {
+  state = {
+    open: false,
+  };
+
+  openModal = () => {
+    this.setState({
+      open: true,
+    });
+  };
+
+  closeModal = () => {
+    this.setState({
+      open: false,
+    });
+  };
+
+  onClickTrigger = e => {
+    const { trigger } = this.props;
+    const { open } = this.state;
+
+    if (trigger.props.onClick) {
+      trigger.props.onClick(e);
```

Comment:
Accessing props like this is a big no-no. A trigger should only be responsible for opening modal via `onClick`. If a user wants to specify any `onOpen` or `onClose` logic, then it should be passed as a callback to `Modal` and called after updating state